promql/functions.go

```diff
@@ -127,6 +127,49 @@ func funcIncrease(ev *evaluator, args Expressions) model.Value {
 	return funcDelta(ev, args).(vector)
 }
 
+// === irate(node model.ValMatrix) Vector ===
+func funcIrate(ev *evaluator, args Expressions) model.Value {
+	resultVector := vector{}
+	for _, samples := range ev.evalMatrix(args[0]) {
+		// No sense in trying to compute a rate without at least two points. Drop
+		// this vector element.
+		if len(samples.Values) < 2 {
+			continue
+		}
+
+		var currentSample, previousSample model.SamplePair
+		for _, sample := range samples.Values {
```

Comment:
Hmm, I'd tried that first time around and couldn't get it working. Works now though.